src/librustdoc/lib.rs

```diff
@@ -230,6 +234,10 @@ pub fn main_args(args: &[String]) -> isize {
         }
     };
 
+    if let Some(playground) = matches.opt_str("playground-url") {
+        html::markdown::PLAYGROUND.with(|s| { *s.borrow_mut() = Some((None, playground)); });
```

Comment:
When doing HTML rendering the crate name needs to be set so the correct `extern crate` can be added to the code. The first item in the tuple is the crate name.